src/Time/Formatting.hs

```diff
@@ -62,13 +62,13 @@ instance (KnownRatName unit, Series units)
 #else
     seriesF t = let newUnit = toUnit @unit t
 #endif
-                    format  = floorUnit newUnit
-                    timeStr = case floor newUnit :: Int of
+                    flooredNewUnit  = floorUnit newUnit
+                    timeStr = case flooredNewUnit of
                                    0 -> ""
-                                   _ -> show format
-                in if t == 0 || format == newUnit
+                                   _ -> show flooredNewUnit
+                in if t == 0 || flooredNewUnit == newUnit
                    then show newUnit
-                   else timeStr ++ seriesF @units @unit (newUnit - format)
+                   else timeStr ++ seriesF @units @unit (newUnit - flooredNewUnit)
```

Comment:
I think that instead of pattern-matching on `flooredUnit` we can pattern-match on `newUnit - flooredUnit`. I think the following holds:

* `floor newUnit == 0 ≡ newUnit - floor newUnit == 0`
* `newUnit == 0 ==> newUnit - floorUnit == 0`

So we can pattern match on the result of subtraction simplify case. Well, by simplifying I also mean remove `t == 0` check. But I don't think that `seriesF` function would be a bottleneck of any application. It's already uses `String` and polymorphic recursion. And by removing check `t == 0` we only decrease performance of showing zero time while increasing performance of showing non-zero time. I think `seriesF` would be called on non-zero times more often 🙂